packages/block-library/src/navigation/edit/index.js

```diff
@@ -454,6 +462,27 @@ function Navigation( {
 		{ open: overlayMenuPreview }
 	);
 
+	// Set menu to published if it's a draft on site load in order to make the post dirty.
+	useEffect( async () => {
+		hideMenuAutoPublishDraftNotice();
+		if ( ! isDraftNavigationMenu ) return;
+		try {
+			await editEntityRecord(
+				'postType',
+				'wp_navigation',
+				navigationMenu?.id,
+				{
+					status: 'publish',
```

Comment:
In this case:
 - User creates a classic menu from navigation.
 - User reloads or exits without saving.
 - User comes back to the site editor, it appears the classic menu (as a draft)

There is no new ref making the post dirty. What could be the best way to make the editor dirty?

Reply:
So the problem would be that 

- Site Editor shows draft classic menu
- Front of site shows whatever was there before

That's because the changes to the _post_ in which the Navigation block resides have yet to be saved right?

Reply:
I understand now the idea is to prompt the user to publish the menu they have set as a draft. We should add this as the reason we make the post dirty in the comment. That's all.